src/lib/util/printer.ml

```diff
@@ -0,0 +1,204 @@
+(******************************************************************************)
+(*                                                                            *)
+(*     Alt-Ergo: The SMT Solver For Software Verification                     *)
+(*     Copyright (C) 2018-2020 --- OCamlPro SAS                               *)
+(*                                                                            *)
+(*     This file is distributed under the terms of the license indicated      *)
+(*     in the file 'License.OCamlPro'. If 'License.OCamlPro' is not           *)
+(*     present, please contact us to clarify licensing.                       *)
+(*                                                                            *)
+(******************************************************************************)
+
+open Format
+
+(*********** Colors ***********)
+type style =
+  | Normal
+
+  | Bold
+  | Bold_off
+  | Underline
+  | Underline_off
+
+  | FG_Black
+  | FG_Red
+  | FG_Green
+  | FG_Orange
+  | FG_Blue
+  | FG_Magenta
+  | FG_Cyan
+  | FG_Default
+
+  | BG_Black
+  | BG_Red
+  | BG_Green
+  | BG_Orange
+  | BG_Blue
+  | BG_Magenta
+  | BG_Cyan
+  | BG_Default
+
+(* See https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters
+   for some values *)
+let to_value = function
+  | Normal -> "0"
+
+  | Bold -> "1"
+  | Bold_off -> "22"
+  | Underline -> "4"
+  | Underline_off -> "24"
+
+  | FG_Black -> "30"
+  | FG_Red -> "31"
+  | FG_Green -> "32"
+  | FG_Orange -> "33"
+  | FG_Blue -> "34"
+  | FG_Magenta -> "35"
+  | FG_Cyan -> "36"
+  | FG_Default -> "39"
+
+  | BG_Black -> "40"
+  | BG_Red -> "41"
+  | BG_Green -> "42"
+  | BG_Orange -> "43"
+  | BG_Blue -> "44"
+  | BG_Magenta -> "45"
+  | BG_Cyan -> "46"
+  | BG_Default -> "49"
+
+let style_of_tag t =
+  match Format_shims.get_stag t with
+  | "n" -> Normal
+  | "bold" -> Bold
+  | "/bold" -> Bold_off
+  | "underline" -> Underline
+  | "/underline" -> Underline_off
+
+  | "fg_black" -> FG_Black
+  | "fg_red" -> FG_Red
+  | "fg_green" -> FG_Green
+  | "fg_orange" -> FG_Orange
+  | "fg_blue" -> FG_Blue
+  | "fg_magenta" -> FG_Magenta
+  | "fg_cyan" -> FG_Cyan
+  | "fg_default" -> FG_Default
+
+  | "bg_black" -> BG_Black
+  | "bg_red" -> BG_Red
+  | "bg_green" -> BG_Green
+  | "bg_orange" -> BG_Orange
+  | "bg_blue" -> BG_Blue
+  | "bg_magenta" -> BG_Magenta
+  | "bg_cyan" -> BG_Cyan
+  | "bg_default" -> BG_Default
+
+  | _ -> raise Not_found
+
+let close_tag = function
+  | Bold -> Bold_off
+  | Underline -> Underline_off
+  | FG_Black | FG_Red | FG_Green | FG_Default -> FG_Default
+  | BG_Black | BG_Red | BG_Green | BG_Default -> BG_Default
+  | _ -> Normal
+
+let start_stag t =
+  try Printf.sprintf "\027[%sm" (to_value (style_of_tag t))
```

Comment:
`\027` can be misinterpreted when written directly, let's use hexadecimal representations for our sake
```suggestion
  try Printf.sprintf "\x1B[%sm" (to_value (style_of_tag t))
```